sympy/polys/polytools.py

```diff
@@ -6838,7 +6826,8 @@ def _new(cls, basis, options):
 
     @property
     def args(self):
-        return (Tuple(*self._basis), Tuple(*self._options.gens))
+        basis = (p.as_expr() for p in self._basis)
```

Comment:
@oscarbenjamin I'm guessing this is where the performance regression issues come from, though I didn't verify it. Would it help to cache this?

Reply:
I wonder if we should use `cached_property` generally now that it's in the stdlib:
https://docs.python.org/dev/library/functools.html?highlight=s#functools.cached_property